substrate/dex/pallet/src/lib.rs

```diff
@@ -154,11 +154,34 @@ pub mod pallet {
   #[pallet::storage]
   pub type Pools<T: Config> = StorageMap<_, Blake2_128Concat, PoolId, PoolInfo<Coin>, OptionQuery>;
 
+  #[pallet::storage]
+  #[pallet::getter(fn last_quote_for_block)]
+  pub type LastQuoteForBlock<T: Config> =
+    StorageDoubleMap<_, Identity, BlockNumberFor<T>, Identity, Coin, [u8; 8], ValueQuery>;
+
+  /// Moving window of oracle prices.
+  #[pallet::storage]
+  #[pallet::getter(fn oracle_prices)]
+  pub type OraclePrices<T: Config> = StorageMap<_, Identity, [u8; 8], u16, OptionQuery>;
+  impl<T: Config> Pallet<T> {
+    /// Get the highest sustained(lowest) value of the window
+    pub fn highest_sustained_price() -> Amount {
```

Comment:
Should we have a TODO here to consider an algorithm which removes outliers? This algorithm might work a good bit better if we remove the bottom `n` values (so some value sustained over 90% of blocks instead of all blocks in the window), 

I don't want to force that economic discussion into this PR though.